tests/test_interpolation.py

```diff
@@ -709,3 +713,31 @@ def test_empty_stack() -> None:
     """
     with pytest.raises(GrammarParseError):
         grammar_parser.parse("ab}", lexer_mode="VALUE_MODE")
+
+
+@pytest.mark.parametrize("ref", ["missing", "invalid"])
+def test_invalid_intermediate_result_when_not_throwing(
+    ref: str, restore_resolvers: Any
+) -> None:
```

Comment:
Is this test still valuable enough after the recent changes to exceptions handling?

Reply:
I think it's good to keep it to ensure that future changes do not accidentally "leak" unintended missing/None inputs to resolvers. How valuable that is -- I'm not sure... happy to remove it if you prefer.